Sources/Public/CalendarView.swift

```diff
@@ -990,6 +997,15 @@ extension CalendarView: UIScrollViewDelegate {
     }
   }
 
+  public func scrollViewShouldScrollToTop(_ scrollView: UIScrollView) -> Bool {
+    if content.monthsLayout.isHorizontal {
+      return false
+    } else {
+      self.scrollView.isAboutToScrollToTopFromSystemGesture = true
+      return true
```

Comment:
Instead would it be more direct to return false here and trigger the programatic scrolling right here?

Reply:
That was my initial thinking (and it would definitely make things a bit simpler), but I'm worried that this function could be called / checked at other times by the system. This really should be a side-effect-free way for the system to check whether a scroll view supports the scroll-to-top gesture, and I have no idea if this can be called in other situations no or in the future. Thinking the current approach is a bit more resilient.

Reply:
Although I guess you could argue that if this was called in a situation where we weren't about to scroll to the top, then the next call to `scrollView.contentOffset`'s setter would also be incorrect... that wouldn't have too bad of an affect - we'd just ignore the first call to the setter. Maybe the contentOffset setter override should also check the target offset to make sure it actually is the top of the scroll view.

Reply:
So the logic would then be:

1. `scrollViewShouldScrollToTop` called, set `isAboutToScrollToTopFromSystemGesture` to `true`
2. `scrollView.contentOffset.set` called, cancel the scroll only if the target offset y is actually at the top
3. set `isAboutToScrollToTopFromSystemGesture` to `false`

Reply:
To me, assuming that the scroll to top tap triggers the scroll view's `contentOffset` setter seems less reliable than assuming that there is a 1:1 relationship between `scrollViewShouldScrollToTop` getting called and the scroll to top gesture. I think returning false from the `scrollViewShouldScrollToTop` function and triggering your custom action is the typical way of customizing this behavior despite the naming which suggests it shouldn't have side-effects. I've used this approach before. And if it's common, Apple would be less likely to change the behavior of it down the road breaking a lot of apps.

Reply:
I see what you're saying - either way we need to make an assumption somewhere about how Apple is calling into these things. My initial concern was that someone else could actually call this property / function from their app, but since our scroll view is a private implementation detail of the calendar, I don't think we actually need to worry about this.

Latest commit has some changes!